Add rt700 i3c support on cm33_cpu1 core

lucien-nxp opened the pull request:
Add support for p3t1755 temperature sensor on RT700 cm33_cpu1 core

lucien-nxp committed:
dts: arm: nxp: add i3c2/3 instances for cm33_cpu1

add i3c2/3 instances for cm33_cpu1

Signed-off-by: Lucien Zhao <lucien.zhao@nxp.com>

lucien-nxp committed:
sample: sensor: Add support for temperature sensor on RT700 cm33_cpu1

Add support for p3t1755 temperature sensor on RT700 cm33_cpu1 core
Add net driver condition to exclude rt700 platform

Signed-off-by: Lucien Zhao <lucien.zhao@nxp.com>

MaureenHelm commented on samples/sensor/thermometer/boards/mimxrt700_evk_mimxrt798s_cm33_cpu1.overlay:
```diff
@@ -0,0 +1,27 @@
+
+#include <freq.h>
+
+/ {
+	aliases {
+		ambient-temp0 = &p3t1755;
+	};
+};
+
+&i3c3 {
+	status = "okay";
+
+	i2c-scl-hz = <DT_FREQ_K(400)>;
+	i3c-scl-hz = <DT_FREQ_M(4)>;
+	i3c-od-scl-hz = <DT_FREQ_K(1500)>;
+
+	p3t1755: p3t1755@4800000236152a0090 {
+		compatible = "nxp,p3t1755";
+		reg = <0x48 0x0236 0x152a0090>;
+		status = "okay";
+	};
+};
```
No need for an overlay, this can go in the board's dts

lucien-nxp replied:
Move content from .overlay to board's dts. Thank you

lucien-nxp replied:
ambient-temp0 still need to be defined in .overlay. Otherwise secure_mqtt_sensor_actuator will be included into CI because the filter condition is "ambient-temp0".
`tests:
  sample.net.secure_mqtt_sensor_actuator:
    harness: net
    tags:
      - net
      - mqtt
      - sensors
    filter: dt_alias_exists("ambient-temp0")
    integration_platforms:
      - adi_eval_adin1110ebz`

MaureenHelm replied:
>  Otherwise secure_mqtt_sensor_actuator will be included into CI because the filter condition is "ambient-temp0".

Why is this a problem?

lucien-nxp replied:
At first, I moved all overlay related content into the board level, but CI could not compile successfully on "secure_mqtt_sensor_actuator" case  because this case was not supported now, but this case filter only whether there was any ambient_temp0 enabled.
![image](https://github.com/user-attachments/assets/db603aee-fa8f-4e68-8570-662ae58f00de)

lucien-nxp commented:
Blocked by #88305, waiting for merge.

MaureenHelm commented on samples/sensor/thermometer/sample.yaml:
```diff
@@ -35,6 +35,7 @@ tests:
         - "Temperature is [0-9|.]+°C"
     integration_platforms:
       - frdm_mcxn947/mcxn947/cpu0 # p3t1755
+      - mimxrt700_evk/mimxrt798s/cm33_cpu1 # p3t1755
```
The p3t1755 sensor is already covered in the previous integration platform, don't need to add another integration platform for the same sensor.

lucien-nxp replied:
Deleted.

lucien-nxp commented on samples/sensor/thermometer/boards/mimxrt700_evk_mimxrt798s_cm33_cpu1.overlay:
```diff
@@ -0,0 +1,11 @@
+
+/ {
+	aliases {
+		ambient-temp0 = &p3t1755;
```
Move to board level and modify the filter.

MaureenHelm reviewed (commented):
Code looks good, but please move the sample filtering changes to a separate commit

lucien-nxp committed:
samples: net/secure_mqtt_sensor_actuator: replenish filter condition

The original filtering condition was only whether ambient-temp0
device existed or not. However, registering ambient temp0 does
not mean that NET-related devices have been enabled.

Signed-off-by: Lucien Zhao <lucien.zhao@nxp.com>

lucien-nxp commented:
Hi @dleach02 @decsny 
Could you help review the PR?
